### ISSUE TITLE
Please improve DocFX documentation

### ISSUE THREAD

**Issue by nathan-alden-sr** [dotnet/docfx#5723]
It seems strange that a project whose purpose is to build documentation itself has such [poor documentation](https://dotnet.github.io/docfx/tutorial/docfx_getting_started.html). The DocFX tool and concept seem very cool and flexible; however, there is apparently no official documentation describing the architecture of DocFX--what each concept means, how they relate, etc. (documentation about the domain model of the tool)--nor is there detailed documentation about the `docfx.json` configuration file (yes, there is a JSON schema but that is too fine-grained to be of much use).

I was told there is a v3 of DocFX underway. Please consider greatly enhancing DocFX documentation to better introduce new folks to the tool and its model. Adoption will likely increase greatly if you do this.

**Comment by grokys** [comment]
PLEASE! I've just come to docfx again after several years and disappointed to find the documentation as obtuse as ever. @nathan-alden-sr correctly points to the `docfx.json` documentation being bad, but the documentation for theming is even worse.

**Comment by mdschweda** [comment]
Agreed. The state of the documentation is quite ironic for a documentation tool.

**Comment by AZ-X** [comment]
Quite agreed. I just narrow down to a sub project called ‘MarkdownLite’.  The codes inside this project are strange to me, while pattern’s names are superficial in everywhere.
